src/Common/PlatformLinux.inl

```diff
@@ -1108,3 +1108,29 @@ inline tm* localtime_safe(const time_t *time, struct tm* result){ return localti
 #define xr_strerror(errno, buffer, bufferSize) strerror_r(errno, buffer, sizeof(buffer))
 
 using xrpid_t = pid_t;
+
+// This is a drop-in replacement for calls to linux 'read' function. We use this because unlike other OSes
+// Linux 'read' function can return less then the requested number of bytes and might need to be called multiple times.
+// Apart from this, it behaves exactly as you would expect and matches the other OSes implementation.
+// See also: https://www.man7.org/linux/man-pages/man2/read.2.html
+inline ssize_t xr_read(int file_handle, void *buffer, size_t count)
```

Comment:
Actually it make sense to make this implementation common, so any new platform with such behaviour will be handled automatically. And it will not harm other platforms.

Reply:
Alright where would be a good place to put this? Directly into Platform.h?